plugins/memcached/CMakeLists.txt

```diff
@@ -25,27 +25,11 @@
 project( memcached )
 
 if (MEMCACHED)
-    ADD_PLUGIN(memcached)
-    if(MAKE_MEMCACHED)
-        # Currently libmemcached versions are not sufficient on ubuntu 12.04 and 14.04 LTS
-        # until then, we build the required libraries from source
-
+    if(NOT DEFINED LIBMEMCACHED_VERSION)
         set(LIBMEMCACHED_VERSION "1.0.10")
-        add_custom_command(
-            OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/libmemcached/.libs/libmemcached.so
-            COMMAND wget -q https://launchpad.net/libmemcached/1.0/${LIBMEMCACHED_VERSION}/+download/libmemcached-${LIBMEMCACHED_VERSION}.tar.gz
-            COMMAND tar -xzf ${CMAKE_CURRENT_BINARY_DIR}/libmemcached-${LIBMEMCACHED_VERSION}.tar.gz
-            COMMAND cd ${CMAKE_CURRENT_BINARY_DIR}/libmemcached-${LIBMEMCACHED_VERSION}
-            COMMAND ${CMAKE_CURRENT_BINARY_DIR}/libmemcached-${LIBMEMCACHED_VERSION}/configure > /dev/null
-            COMMAND make -j > /dev/null
-            WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
-            COMMENT "fetching and building libmemcached-${LIBMEMCACHED_VERSION}")
-        add_custom_target(generate-libmemcached
-            DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/libmemcached/.libs/libmemcached.so)
-        add_library(libmemcached SHARED IMPORTED)
-        set_target_properties(libmemcached
-            PROPERTIES IMPORTED_LOCATION ${CMAKE_CURRENT_BINARY_DIR}/libmemcached/.libs/libmemcached.so.11.0.0)
-        set(LIBMEMCACHED_INCLUDE_DIR ${CMAKE_CURRENT_BINARY_DIR}/libmemcached-${LIBMEMCACHED_VERSION})
+    endif()
+    ADD_PLUGIN(memcached PACKAGES LIBMEMCACHED MINVERSION ${LIBMEMCACHED_VERSION})
```

Comment:
LIBMEMCACHED_VERSION being specified by the builder at config, should still be checked to be greater or equal to the actual min version rather than being passed in as the min version, right?